### PR TITLE
Remove vm_event and replace UMC::Semaphore with condition variable

### PR DESCRIPTION
**Pull request by aobolensk**

### CODE REVIEW

File: _studio/mfx_lib/scheduler/include/mfx_scheduler_core.h [hunk 1]
```diff
@@ -32,9 +32,6 @@
 // synchronization stuff
 #include <vm_time.h>
 
-#include <umc_semaphore.h>
-#include <umc_event.h>
-
 #include <vector>
```
Comment (dvrogozh) [inline _studio/mfx_lib/scheduler/include/mfx_scheduler_core.h:35]: Please, break into 2 commits where you remove instances one by one.

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: _studio/mfx_lib/scheduler/include/mfx_scheduler_core.h (hunk 1 above)
Comment (aobolensk) [reply _studio/mfx_lib/scheduler/include/mfx_scheduler_core.h:35]: Done

File: _studio/mfx_lib/scheduler/linux/src/mfx_scheduler_core_task_management.cpp [hunk 2]
```diff
@@ -593,7 +593,8 @@ void mfxSchedulerCore::MarkTaskCompleted(const MFX_CALL_INFO *pCallInfo,
     // wake up external threads waiting for a free task object
     if (taskReleased)
     {
-        m_freeTasks.Signal(1);
+        ++m_freeTasksCount;
```
Comment (dvrogozh) [inline _studio/mfx_lib/scheduler/linux/src/mfx_scheduler_core_task_management.cpp:596]: can you, please, double check that this is being executed under mutex?

File: _studio/mfx_lib/CMakeLists.txt [hunk 3]
```diff
@@ -71,7 +71,7 @@ append_property( mfx_common_hw COMPILE_FLAGS " -DMFX_RT" )
 
 ### libmfxhw
 list( APPEND mdirs
-  scheduler vpp decode/vc1
+  scheduler/linux vpp decode/vc1
```
Comment (dvrogozh) [inline _studio/mfx_lib/CMakeLists.txt:74]: Build for the "move" patch failed. We need to have patches buildable independently. Please, make a year copyright change here in this commit.

File: _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2019 Intel Corporation
+// Copyright (c) 2018-2020 Intel Corporation
```
Comment (dvrogozh) [inline _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h:1]: "event" commit - change title:
```
scheduler: remove vm_event
```

File: _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h [hunk 5]
```diff
@@ -384,8 +383,9 @@ class mfxSchedulerCore : public MFXIScheduler2
 
 
 
-    // Event to wait free task objects
-    UMC::Semaphore m_freeTasks;
+    // Condition variable to wait free task objects
```
Comment (dvrogozh) [inline _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h:386]: "semaphore" commit - change title:
```
scheduler: remove UMC::Semaphore
```

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: _studio/mfx_lib/scheduler/linux/src/mfx_scheduler_core_task_management.cpp (hunk 2 above)
Comment (aobolensk) [reply _studio/mfx_lib/scheduler/linux/src/mfx_scheduler_core_task_management.cpp:597]: Fixed

File: _studio/mfx_lib/CMakeLists.txt (hunk 3 above)
Comment (aobolensk) [reply _studio/mfx_lib/CMakeLists.txt:74]: Changed copyright years for moved files

File: _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h (hunk 4 above)
Comment (aobolensk) [reply _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h:1]: Done

File: _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h (hunk 5 above)
Comment (aobolensk) [reply _studio/mfx_lib/scheduler/linux/include/mfx_scheduler_core.h:386]: Done

**Review by dvrogozh — DISMISSED** [review]

**Review by dvrogozh — APPROVED** [review]

**Review by onabiull — APPROVED** [review]

**Pull request merged**